colbuilder: explicitly list all processors in canWrap

yuzefovich opened the pull request:
This commit hardens the processor core check in `canWrap` to ensure that
all possible processor cores are explicitly listed there. There were
multiple cores missing from the switch, and for all of them we now return
an error (meaning that flows with any of them won't be vectorized, as
before this change). There is no fundamental reason for why most of these
prohibit vectorization, rather it seems not worth it. Plus, there could
be some subtle changes in behavior because colflow infra behaves
differently that rowflow one (e.g. when facing a network interruption or
an error on one of the nodes).

The main goal of the commit is to ensure we won't forget this type switch
when adding a new processor implementation and will be forced to decide
whether to allow vectorization or not.

Epic: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/145547)
<!-- Reviewable:end -->

yuzefovich committed:
colbuilder: explicitly list all processors in canWrap

This commit hardens the processor core check in `canWrap` to ensure that
all possible processor cores are explicitly listed there. There were
multiple cores missing from the switch, and for all of them we now return
an error (meaning that flows with any of them won't be vectorized, as
before this change). There is no fundamental reason for why most of these
prohibit vectorization, rather it seems not worth it. Plus, there could
be some subtle changes in behavior because colflow infra behaves
differently that rowflow one (e.g. when facing a network interruption or
an error on one of the nodes).

The main goal of the commit is to ensure we won't forget this type switch
when adding a new processor implementation and will be forced to decide
whether to allow vectorization or not.

Release note: None

DrewKimball reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Nice!

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/145547#-:-OPc23U4Cw8zyUnnFzy8:b-8gk8uj)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)



<!-- Sent from Reviewable.io -->
